docs/csharp/language-reference/operators/patterns.md

```diff
@@ -110,6 +112,24 @@ As the preceding example shows, you can repeatedly use the pattern combinators i
 
 :::code language="csharp" source="snippets/patterns/PatternCombinators.cs" id="WithParentheses":::
 
+## Property pattern
+
+Beginning with C# 8.0, you use a *property pattern* to match expression's properties or fields against nested patterns, as the following example shows:
+
+:::code language="csharp" source="snippets/patterns/PropertyPattern.cs" id="BasicExample":::
+
+A property pattern matches an expression when an expression result is non-null and every nested pattern matches the corresponding property or field of the expression result.
+
+You can also add a runtime type check and a variable declaration to a property pattern, as the following example shows:
+
+:::code language="csharp" source="snippets/patterns/PropertyPattern.cs" id="WithTypeCheck":::
+
+As you can use any pattern as a nested pattern in a property pattern, you can perform *recursive* pattern matching. That is, you can match parts of data against nested patterns, as the following example shows:
```

Comment:
ping @MadsTorgersen 

As we keep working on this, I want to consistent in our use of "nested" and "recursive" patterns. I don't think we are now, and I'm not confident in the distinction.

Reply:
I understand that in the following way: 1. "recursive" refers to the "pattern matching" combo, not to "pattern". "Recursive pattern" doesn't sound to me yet. 2. Nested patterns can be also called as "subpatterns".

Reply:
Here's how I've been thinking of it:
- _Recursive patterns_ are a term for the _kinds_ of patterns that contain other patterns; e.g. positional, property, relational and logical patterns.
- _Nested patterns_ are concrete occurrences in code where a pattern is nested within another pattern.

I.e. "recursive pattern" classifies a language feature, "nested pattern" describes code.

I don't have deeply held convictions here. That's just where my mind goes to with these terms.

Reply:
This paragraph matches Mads' definitions (or my current understanding). This looks great.